lib/src/piece/assign.dart

```diff
@@ -19,37 +19,75 @@ import 'piece.dart';
 ///
 ///     var x = 123;
 ///
-/// If the value is a delimited "block-like" expression, then we allow splitting
-/// inside the value but not at the `=` with no additional indentation:
+/// This state also allows splitting the right side if it can be block
+/// formatted:
 ///
 ///     var list = [
 ///       element,
 ///     ];
 ///
-/// [_atOperator] Split after the `=`:
+/// [_blockSplit] Block split in the operands that support it and expression
+/// split in the others. This state requires at least one operand to support
+/// block splitting. Examples:
 ///
-///     var name =
-///         longValueExpression;
+///     var [
+///       element,
+///     ] = value;
+///
+///     var [
+///       element,
+///     ] = longOperand +
+///         anotherOperand;
+///
+///     var (longVariable &&
+///         anotherVariable) = [
+///       element,
+///     ];
+///
+/// [_atOperator] Split at the `=` or `in` operator and allow expression
```

Comment:
I don't know the exact ruling on this, but would it be better to have these doc comments at the site of the member? (below) I find myself scrolling up to look at an example.

Reply:
> I don't know the exact ruling on this

Yeah, I've gone back and forth on this too. I could never decide if it makes more sense to document the different styles all in one place on the class, at the declaration of each `State` field, or at the top of the `format()` method.

I've tried to consistently document them on the Piece class itself. That's mainly because most pieces use `State.unsplit` and often `State.split` for some style, and there's no other obvious place to document how the piece styles itself when in those states.